rust/lance/src/index.rs

```diff
@@ -550,6 +555,7 @@ impl DatasetIndexExt for Dataset {
             "num_indexed_rows": num_indexed_rows,
             "num_unindexed_fragments": num_unindexed_fragments,
             "num_unindexed_rows": num_unindexed_rows,
+            "num_indexed_rows_per_delta": num_indexed_rows_per_delta,
```

Comment:
Is this implicitly in the same order that `list_indices` returns? Do we even guarantee an order there?

Reply:
I think we guarantee append order, let me check